web/src/pill_typeahead.ts

```diff
@@ -64,54 +89,76 @@ export function set_up($input, pills, opts) {
             }
             return source;
         },
-        highlighter_html(item, query) {
-            if (include_streams(query)) {
+        highlighter_html(item: TypeaheadItem, query: string): string {
+            if (include_streams(query) && item.type === "stream") {
                 return typeahead_helper.render_stream(item);
             }
 
-            if (include_user_groups && user_groups.is_user_group(item)) {
+            if (
+                include_user_groups &&
+                item.type === "user_group" &&
+                user_groups.is_user_group(item)
```

Comment:
I think it should be fine to remove `user_groups.is_user_group`, since the user group items are all created explicitly with user groups, but wanted to check in.

Reply:
Yeah, that sounds right.